microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsCdiExtension.java

```diff
@@ -449,6 +462,21 @@ private void recordAnnotatedGaugeSite(@Observes @WithAnnotations(Gauge.class) Pr
             return;
         }
 
+        Annotation annotation = type.getAnnotation(RequestScoped.class);
+        if (annotation != null) {
+            errors.fatal(clazz, "Cannot configure @Gauge on a request scoped bean");
```

Comment:
Is failing preferable to warning in this case?

Reply:
See: https://github.com/eclipse/microprofile-metrics/pull/367/files

In general disabling a feature means it works differently from what user wanted - in this case it cannot work and the annotation is wrong, so I think an exception is the right way to process it.